### ISSUE TITLE
Bad predictions of the models

### ISSUE THREAD

**Issue by 404akhan** [songyouwei/ABSA-PyTorch#26]
Try to input first:
`trump is good, but obama is bad`, with `trump` being target.
Then, second with `obama` being target. 
None of the models achieve the result they claim to achieve, because they output the same sentiment for both sentences. 
(Sorry if wrongly tested, but i think i didn't).

**Comment by GeneZC** [comment]
I think you are doing the right thing, and you could refer to issue #25.
Here I try to give you two reasons why most models suffer from the problem you mentioned:
Firstly,  you are testing models with sentences from the domain which they never met with in training dataset, i.e., in laptop or restaurant domain;
Secondly,  models occasionally make wrong prediction in such samples as they don't consider the relationships between different aspects within one sentence.
From my perspective, above-mentioned two flaws are the directions we should concentrate in later works.

**Comment by 404akhan** [comment]
I trained models on twitter dataset 
` https://github.com/songyouwei/ABSA-PyTorch/blob/master/datasets/acl-14-short-data/train.raw `
The dataset has both Obama and Trump. So the first reason doesn't apply.
As for the second reason, I remember reading papers on this topic and the first thing they tell is that target-dependent sentiment analysis should identify these kind of scenarios and handdle them. But, after that authors turn off their brain, throw their neural networks at data and even forget what they actually wanted. So if current models cannot identify this basic scenario then they are just practically useless. And the only reason those papers get published is because their "accuracy" is better than the previous meaningless model. 
Correct me if I am wrong.
